uninstall.sh

```diff
@@ -0,0 +1,290 @@
+      [ "$OS_VER_MAJOR" == "10" ] && SUPPORTED=true
+      ;;
+    centos)
+      [ "$OS_VER_MAJOR" == "7" ] && SUPPORTED=true
+      [ "$OS_VER_MAJOR" == "8" ] && SUPPORTED=true
+      ;;
+  esac
+
+  # exit if not supported
+  if [ "$SUPPORTED" == true ]; then
+    echo "* $OS $OS_VER is supported."
+  else
+    echo "* $OS $OS_VER is not supported"
+    error "Unsupported OS"
+    exit 1
+  fi
+}
+
+### Main uninstallation functions ###
+
+rm_files(){
+  if [ "$RM_PANEL" == true ]; then
+    rm -rf /var/www/pterodactyl
+    [ "$OS" != "centos" ] && unlink /etc/nginx/sites-enabled/pterodactyl.conf
+    [ "$OS" != "centos" ] && rm -f /etc/nginx/sites-available/pterodactyl.conf
+    [ "$OS" == "centos" ] && rm -f /etc/nginx/conf.d/pterodactyl.conf
+  fi
+
+  [ "$RM_WINGS" == true ] && rm -rf /etc/pterodactyl /usr/local/bin/wings
+}
+
+rm_services(){
+  systemctl disable --now mariadb
+  systemctl disable --now pteroq
+  rm -rf /etc/systemd/system/pteroq.service
+  case "$OS" in
+    debian | ubuntu)
+      systemctl disable --now redis-server
+      ;;
+    centos)
+      systemctl disable --now redis
+      systemctl disable --now php-fpm
+      rm -rf /etc/php-fpm.d/www-pterodactyl.conf
+      ;;
+  esac 
+}
+
+rm_cron(){
+  # shellcheck disable=SC2063
+  crontab -l | grep -v "* * * * * php /var/www/pterodactyl/artisan schedule:run >> /dev/null 2>&1" | crontab -
+}
+
+rm_database(){
+  valid_db=$(mysql -u root -p -e "SELECT schema_name FROM information_schema.schemata;" | grep -v -E -- 'schema_name|information_schema|performance_schema')
+  warning "Be careful! This database will be deleted!"
+  if [[ "$valid_db" == *"panel"* ]]; then
+    echo -n "* Database called panel has been detected. Is it the pterodactyl database? (y/N): "
+    read -r is_panel
+    [[ "$is_panel" =~ [Yy] ]] && DATABASE=panel || echo "$valid_db"
+  else
+    echo "$valid_db"
+  fi
+  while [ -z "$DATABASE" ] || [[ $valid_db != *"$database_input"* ]]; do
+    echo -n "* Choose the panel database(to skip type none): "
+    read -r database_input
+    DATABASE=$database_input
+  done
+  [[ "$DATABASE" != "none" ]] && mysql -u root -p -e "DROP $DATABASE"
+  # Exclude usernames User and root (Hope no one uses username User)
+  valid_users=$(mysql -u root -p -e "SELECT user FROM mysql.user;" | grep -v -E -- 'User|root')
+  warning "Be careful! This user will be deleted!"
+  if [[ "$valid_users" == *"pteroactyl"* ]]; then
+    echo -n "* User called pterodactyl has been detected. Is it the pterodactyl user? (y/N): "
+    read -r is_user
+    [[ "$is_user" =~ [Yy] ]] && USER=pterodactyl || echo "$valid_users"
+  else
+    echo "$valid_users"
+  fi
+  while [ -z "$USER" ] || [[ $valid_users != *"$user_input"* ]]; do
+    echo -n "* Choose the panel user(to skip type none): "
+    read -r user_input
+    USER=$user_input
+  done
+  [[ "$USER" != "none" ]] && mysql -u root -p -e "DROP USER '$USER'@'127.0.0.1'"
+  mysql -u root -p -e "FLUSH PRIVILEGES;"
+}
```

Comment:
@vilhelmprytz Do you think database removal part is good enough to not confuse users?

Reply:
I am skepictal about removing all users but `root`. An ignorant user could easily accidentally wipe databases and users they do not wish to remove. Would it be possible to follow the same kind of "prompt format" as `install-panel.sh`? i.e. ask for MySQL username and database with the default values `panel` and `pterodactyl`.

Also, the uninstall script should be very clear about the fact that it is only meant to be run on systems that used the installation script in the first place. Haven't looked at your code yet so I don't know if you've made this apparent already. We could make the panel script create a file somewhere on the system when the install completes indicating that the system was installed using pterodactyl-installer and not with any other script or manually.

Reply:
No it doesn't delete all users. It gets a list of users, excludes `root` because it can't be deleted. Then it searches for user pterodactyl in that list and if it is there it asks if it is the pterodactyl user and if answer is yes it deletes it. If answer is no it let's the user choose the user to be deleted from the list. Exactly the same is done with database.

I was thinking something better. A file with variables inside which indicate how much of the script has ran and based on that how much to remove. Most users run the script, then fails on something like database user name or pterodactyl admin account username and they need to wipe their system. This uninstall script is for those people.